.gitlab/workflows.yml

```diff
@@ -46,28 +41,98 @@ Python Wheels:
+          for wheel in `ls cuda_quantum-*-cp310-cp310-manylinux_*_$(uname -m).whl 2> /dev/null`; do
+            echo "Adding wheel $wheel."
+            mv $wheel "/tmp/wheels/$wheel"
+          done
+          cd .. && rm -rf _pycudaq*
+        fi
+      done
+    - export python=python3.10 
+    - cd /tmp/wheels/ && $python -m pip install cuda_quantum-*-cp310-cp310-manylinux_*_$(uname -m).whl
+    - cudaq_location=`$python -m pip show cuda-quantum | grep -e 'Location:\ .*$'`
+    - export CUDAQ_INSTALL_PREFIX="${cudaq_location#Location:\ }"
+    - cp -R $CUDAQ_INSTALL_PREFIX/lib64/cmake/fmt $CUDAQ_INSTALL_PREFIX/lib/cmake
+    - export EXATN_COMMIT=9c053ac1671c041827fee3aeb33cfc6bf43762eb
+    - export CUDAQ_TENSORNET_COMMIT=17dc530c4838b2e32125d427feabfd32c1fdd0fc
+    - export CUDAQ_MGMN_SVSIM_COMMIT=e8a6702a222c1a5f3fead4dc5efb11953967dd90
+    # Tensornet Simulator
+    - cuquantum_location=`$python -m pip show cuquantum-cu11 | grep -e 'Location:\ .*$'`
+    - echo $cuquantum_location
+    - export CUQUANTUM_INSTALL_PREFIX="${cuquantum_location#Location:\ }/cuquantum"
+    - export CUQUANTUM_ROOT=$CUQUANTUM_INSTALL_PREFIX 
+    - export CUTENSOR_ROOT="${cuquantum_location#Location:\ }/cutensor"
+    - ln -s $CUQUANTUM_INSTALL_PREFIX/lib/libcustatevec.so.1 $CUQUANTUM_INSTALL_PREFIX/lib/libcustatevec.so 
+    - ln -s $CUQUANTUM_INSTALL_PREFIX/lib/libcutensornet.so.2 $CUQUANTUM_INSTALL_PREFIX/lib/libcutensornet.so 
+    - ln -s $CUTENSOR_ROOT/lib/libcutensor.so.1 $CUTENSOR_ROOT/lib/libcutensor.so
+    - cd $PROJECT_SRCS_DIR
+    - echo "Checking out commit $CUDAQ_TENSORNET_COMMIT on $CI_SERVER_HOST/$CUDAQ_TENSORNET_REPO."
+    - mkdir cudaq_tensornet && cd cudaq_tensornet && git init
+    - git fetch --depth 1 https://$READ_ACCESS_TOKEN:$READ_ACCESS_TOKEN@$CI_SERVER_HOST/$CUDAQ_TENSORNET_REPO $CUDAQ_TENSORNET_COMMIT && git checkout FETCH_HEAD
+    - export PATH=/opt/rh/gcc-toolset-11/root/usr/bin:$PATH
+    - (bash build_wheel.sh $READ_ACCESS_TOKEN && built=true) || built=false
+    - if $built; then `exit 0`; else `exit 1`; fi
+    # MGMN SV  Simulator
+    - cd $PROJECT_SRCS_DIR
+    - echo "Checking out commit $CUDAQ_MGMN_SVSIM_COMMIT on $CI_SERVER_HOST/$CUDAQ_MGMN_SVSIM_REPO."
+    - mkdir cudaq_mgmn_svsim && cd cudaq_mgmn_svsim && git init
+    - git fetch --depth 1 https://$READ_ACCESS_TOKEN:$READ_ACCESS_TOKEN@$CI_SERVER_HOST/$CUDAQ_MGMN_SVSIM_REPO $CUDAQ_MGMN_SVSIM_COMMIT && git checkout FETCH_HEAD
+    - (bash build_wheel.sh $READ_ACCESS_TOKEN && built=true) || built=false
+    - if $built; then `exit 0`; else `exit 1`; fi
+    # Copy artifacts
+    - cd $PROJECT_SRCS_DIR/cuda-quantum
+    - mkdir -p artifacts/wheels_$(uname -m) && cp $PROJECT_SRCS_DIR/cudaq_tensornet/build/libnvqir-tensornet.so $PROJECT_SRCS_DIR/cudaq_tensornet/build/tensornet.config $PROJECT_SRCS_DIR/cudaq_mgmn_svsim/build/libnvqir-nvidia-mgpu.so $PROJECT_SRCS_DIR/cudaq_mgmn_svsim/build/nvidia-mgpu.config artifacts/wheels_$(uname -m)
```

Comment:
Since we've now unified the downstream build scripts, the artifacts should be the same.

Regarding those `.a` static libs, we don't need them to 'use' the backend `.so` libs (`libnvqir-nvidia-mgpu.so` or `libnvqir-tensornet.so`). The needed code should have been pulled in during the linking process. Only those `.so` files that `libnvqir-nvidia-mgpu.so` or `libnvqir-tensornet.so` need are required for runtime usage.

I guess there might be the need for those `.a` files in the docker image for development purposes.
For the wheel build, in particular, the `auditwheel` step will only look for those external `.so` files to pull into the wheel.

Reply:
I don't think anyone using CUDA Quantum should take a direct dependency on the simulator libraries we currently build for the mgpu simulator; they should instead take a dependency only on the cuQuantum packages for development of additional simulators. And nothing in CUDA Quantum other than the tensornet/mgpu simulator needs the static libs, so I think we should remove them also from the docker image.